specutils/spectra/spectrum1d.py

```diff
@@ -442,6 +442,81 @@ def _spectral_slice(self, item):
         reg = SpectralRegion(start, stop)
         return extract_region(self, reg)
 
+    def collapse(self, method, axis=None, physical_type=None):
+        """
+        Collapse the flux array given a method. Will collapse either to a single
+        value (default), over a specified numerical axis or axes if specified, or
+        over the spectral or non-spectral axes if ``physical_type`` is specified.
+
+        If the collapse leaves the spectral axis unchanged, a `~specutils.Spectrum1D`
+        will be returned. Otherwise an `~astropy.units.Quantity` array will be
+        returned.
+
+        Note that these calculations are not currently uncertainty-aware, but do
+        respect masks.
+
+        Parameters
+        ----------
+
+        method : str
+            The method by which the flux will be collapsed. Options are 'mean',
+            'min', 'max', 'sum', and 'median'.
+        axis : int, tuple, optional
+            The axis or axes over which to collapse the flux array.
+        physical_type : str, optional
+            Either 'spectral' or 'spatial'.
+
+        Returns
+        -------
+        :class:`~specutils.Spectrum1D` or :class:`~astropy.units.Quantity`
+
+        """
+        collapse_func = {"mean": np.nanmean, "max": np.nanmax, "min": np.nanmin,
+                         "median": np.nanmedian, "sum": np.nansum}
+
+        if physical_type is not None and axis is not None:
```

Comment:
I'm not sure I see why you wouldn't be able to set each parameter? That is, if I want to collapse over the spectral data, but my spectral data is two-dimensional, why couldn't I use `physical_type = 'spectral'` and `axis = 1`?

Reply:
I wanted to avoid conflicts in the laziest way possible, rather than having to check that the axis is correct given the string input if both are provided. If this is too lazy I can be more robust, but my thought was that if you're inputting the `axis` values you probably already know what axis is what, whereas a user who is a little less familiar with the data structure might find the string input easier to just get out something that makes physical sense.